ambry-store/src/main/java/com.github.ambry.store/IndexValue.java

```diff
@@ -272,12 +273,13 @@ void setFlag(Flags flag) {
    * @param newOffset the new {@link Offset} to be updated for the {@link IndexValue}
    */
   void setNewOffset(Offset newOffset) {
```

Comment:
Like I mentioned earlier, this class mostly acts as a struct, and having this logic within is a bit out of place in my opinion. Especially when the constructor is being used to create a half-baked object. If you must, perhaps what you want is a builder.